### PR TITLE
feat(npm-provenance) - Update to Node.js 20 and add npm package provenance

### PR DESCRIPTION
**Pull request by lilla28**
Add npm package provenance, modified package.json files to contain repository and publish information.
Based on the documentation: https://docs.npmjs.com/generating-provenance-statements#using-third-party-package-publishing-tools using 3rd party libraries (Lerna) to publish we should define EITHER an environment variable (NPM_CONFIG_PROVENANCE) OR add publish configs to the package.json files OR add a .npmrc file where we can set the provenance to true.

Based on the release https://github.com/lerna/lerna/releases/tag/6.6.2 : Lerna does support provenance from the 6.6.2 version. We are on the 8.1.2.

FYI: https://github.com/lerna/lerna/issues/3657 ;

I added package.json reporitory and publishConfigs and environment variable as well to be sure that the provenance flag is set.

### CODE REVIEW

File: src/fdc3/js/composeui-fdc3/package.json [hunk 1]
```diff
@@ -23,6 +23,14 @@
     "jest-environment-jsdom": "^29.7.0",
     "rxjs": "^7.8.1"
   },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/morganstanley/ComposeUI.git"
+  },
+  "publisConfig": {
```
Comment (psmulovics) [inline src/fdc3/js/composeui-fdc3/package.json:30]: Should this not be publishConfig?

Comment (lilla28) [reply src/fdc3/js/composeui-fdc3/package.json:30]: those typos.... thanks will update the PR! :D

File: src/messaging/js/composeui-messaging-client/package.json [hunk 2]
```diff
@@ -16,6 +16,14 @@
         "build:rollup": "rollup -c",
         "test": "jest"
     },
+    "publisConfig": {
```
Comment (kruplm) [inline src/messaging/js/composeui-messaging-client/package.json]: same here

File: src/shell/js/composeui-node-launcher/package.json [hunk 3]
```diff
@@ -10,14 +10,18 @@
...
-    "url": "git+https://github.com/morganstanley/ComposeUI.git#main"
+    "url": "git+https://github.com/morganstanley/ComposeUI.git"
   },
   "scripts": {
     "clean": "rimraf output",
     "build": "npm run clean && tsc",
     "install": "node ./src/cli/install.js",
     "test": "jest"
   },
+  "publisConfig": {
```
Comment (kruplm) [inline src/shell/js/composeui-node-launcher/package.json]: and here

**Review by kruplm — CHANGES_REQUESTED** [review]

File: src/fdc3/js/composeui-fdc3/package.json [hunk 4]
```diff
@@ -23,6 +23,14 @@
     "jest-environment-jsdom": "^29.7.0",
     "rxjs": "^7.8.1"
   },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/morganstanley/ComposeUI.git"
+  },
+  "publishConfig": {
+    "access": "public",
```
Comment (kruplm) [inline src/fdc3/js/composeui-fdc3/package.json:31]: if we don't want to release the composeui-fdc3 package now this should be private. The access is for npm publish, or for any tool that wraps that (in our case Lerna)

**Review by kruplm — CHANGES_REQUESTED** [review]

**Review by kruplm — APPROVED** [review]

**Review by bingenito — APPROVED** [review]

**Pull request merged**